src/c2goto/library/kernel.c

```diff
@@ -107,3 +109,50 @@ unsigned long copy_from_user(void *to, void *from, unsigned long size)
 
   return 0;
 }
+
+void spin_lock_init(spinlock_t *lock)
+{
+  //check if the lock is valid
+  assert(lock != NULL);
+  //initialize the lock
+  lock->locked = 0;
+}
+
+void spin_lock(spinlock_t *lock)
+{
+  //check if the lock is valid
+  assert(lock != NULL);
```

Comment:
You could use __ESBMC_assert(condition, "comment") to provide some informative feedback for the user if the assertion fails.

Reply:
No problem. I have replace the assertion with esbmc_assert and adding reasonbale comment for that